feat(cli): only re-generate provider bindings on get if necessary (#2266)
hashicorp/terraform-cdk: website/docs/cdktf/cli-reference/commands.mdx

This command downloads the providers and modules for an application and
generates CDK constructs for them. It can use the `cdktf.json` configuration file to read the list of providers and modules.

> It only generates currently missing provider bindings, so it is very fast if nothing changed.

Reviewer: ```suggestion
This command only generates currently missing provider bindings, so it is very fast if nothing changed.
```

> If you specify loose version constraints in your `cdktf.json` file and have an existing version already generated locally it will not update the version.
> To update the version you can run `cdktf get --force` to recreate all bindings. If you change a version constraint the `cdktf get` command will recreate the bindings for this provider.

Reviewer: ```suggestion
The command does not update the version when you specify loose version constraints in your `cdktf.json` file and have already generated an existing version locally. To update the version anyway, run `cdktf get --force` to recreate all bindings. 

When you change a version constraint, the `cdktf get` command recreates the bindings for that provider. 
```
